_source/_posts/2019-11-19-secure-kafka-java.md

```diff
@@ -0,0 +1,458 @@
+---
+layout: blog_post
+title: "Kafka with Java: Build a Secure, Scalable Messaging App"
+author: daniel-pereira
+description: "This tutorial shows you how to develop a Java application with Kafka and secure it with OAuth 2.0 and OIDC."
+tags: [java, kafka, spring boot, messaging]
+tweets:
+- "Learn how to integrate @apachekafka in your Java apps and secure everything with @oauth_2."
+- "Kafka is all the rage for messaging! Try it in your @java apps today."
+- "Tutorial: use #Java + #Kafka to implement asynchronous messaging."
+image: 
+---
+
+Today's users expect your app to be accessible from their computer, mobile phone, tablet, or any other device! This transition to Software as a Service (SaaS) as the norm requires developers to effectively integrate with robust tools that scale to handle thousands (or even millions) of requests every second. Apache Kafka is one of the most effective tools for handling those high throughput environments.
+
+In this tutorial, you'll learn the basic concepts behind Apache Kafka and build a fully-functional Java application, capable of both producing and consuming messages from Kafka.
+
+**Prerequisites:** Java 8+, an internet connection, and a [free Okta developer account](https://developer.okta.com/signup).
+
+## A Brief Overview of Apache Kafka
+
+Apache Kafka is a distributed streaming platform that utilizes the publish/subscribe message pattern to interact with applications, and it's designed to create durable messages.
+
+Let's break down those concepts in more detail.
+
+### Distributed Streaming Platform
+
+When you want to run Kafka, you need to start its broker: a simple instance of Kafka running on a machine, just like any other server. The broker is responsible to send, receive, and store messages into the disk.
+
+A single broker is not enough to ensure Kafka can handle a high-throughput of messages. That goal is achieved through many brokers working together at the same time, communicating and coordinating with each other.
+
+A Kafka cluster groups together one or more brokers. Instead of connecting to a single node, your application connects to a cluster that manages all the distributed details for you.
+
+### Publish/Subscribe Messaging System With Durable Messages
+
+The publish/Subscribe is a common pattern in distributed systems. The image below illustrates the basic structure of this pattern within Kafka:
```

Comment:
nit: should publish/subscribe (It's **S**ubscribe above)

Reply:
Fixed.